src/components/flat-table/flat-table.test.tsx

```diff
@@ -0,0 +1,1480 @@
+    <FlatTable width="300px" data-role="ft-wrapper">
+      <FlatTableBody>
+        <FlatTableRow>
+          <FlatTableCell>one</FlatTableCell>
+        </FlatTableRow>
+      </FlatTableBody>
+    </FlatTable>
+  );
+
+  expect(screen.getByTestId("ft-wrapper")).toHaveStyle("width: 300px");
+  expect(screen.getByTestId("flat-table-container")).toHaveStyle(
+    "width: 300px"
+  );
+});
+
+test("should apply the expected `overflowX` styling to the wrapper and container elements when the prop is passed along with `width`", () => {
+  render(
+    <FlatTable overflowX="auto" width="300px" data-role="ft-wrapper">
+      <FlatTableBody>
+        <FlatTableRow>
+          <FlatTableCell>one</FlatTableCell>
+        </FlatTableRow>
+      </FlatTableBody>
+    </FlatTable>
+  );
+
+  expect(screen.getByTestId("ft-wrapper")).toHaveStyle("overflow-x: hidden");
+  expect(screen.getByTestId("flat-table-container")).toHaveStyle(
+    "overflow-x: auto"
+  );
+});
+
+describe("rounded corners are enabled", () => {
+  it("should have the execped border radius styling when no footer is rendered", () => {
+    render(
+      <FlatTable data-role="ft-wrapper">
+        <FlatTableHead>
+          <FlatTableRow>
+            <FlatTableHeader>heading one</FlatTableHeader>
+          </FlatTableRow>
+        </FlatTableHead>
+        <FlatTableBody>
+          <FlatTableRow>
+            <FlatTableCell>child one</FlatTableCell>
+          </FlatTableRow>
+        </FlatTableBody>
+      </FlatTable>
+    );
+    const wrapper = screen.getByTestId("ft-wrapper");
+
+    expect(wrapper).toHaveStyleRule(
+      "border-top-left-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-top-right-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-left-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-right-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-top-left-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `thead ${StyledFlatTableRow}:first-of-type th:first-of-type`,
+      }
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-top-right-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `thead ${StyledFlatTableRow}:first-of-type th:last-of-type`,
+      }
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-left-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `tbody ${StyledFlatTableRow}:last-of-type td:first-child`,
+      }
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-right-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `tbody ${StyledFlatTableRow}:last-of-type td:last-child`,
+      }
+    );
+  });
+
+  it("should override Pager's top border styling so it connects to the table when passed in `footer`,", () => {
```

Comment:
**nitpick:** I _think_ this is the same test as what is on Line 44 of this file.

Reply:
good spot